semgrep/tests/e2e/test_metrics.py

```diff
@@ -52,31 +52,96 @@ def mock_config_request(monkeypatch: MonkeyPatch) -> Iterator[None]:
+            ["--metrics", "auto"],
+            {},
+            True,
+        ),
         ("rules/eqeq.yaml", ["--metrics", "on"], {}, True),
-        ("r/a", ["--metrics", "on"], {}, True),
+        (
+            "r/python.lang.correctness.useless-eqeq.useless-eqeq",
+            ["--metrics", "on"],
+            {},
+            True,
+        ),
         ("rules/eqeq.yaml", ["--metrics", "off"], {}, False),
-        ("r/a", ["--metrics", "off"], {}, False),
+        (
+            "r/python.lang.correctness.useless-eqeq.useless-eqeq",
+            ["--metrics", "off"],
+            {},
+            False,
+        ),
         ("rules/eqeq.yaml", [], {"SEMGREP_SEND_METRICS": "auto"}, False),
-        ("r/a", [], {"SEMGREP_SEND_METRICS": "auto"}, True),
+        (
+            "r/python.lang.correctness.useless-eqeq.useless-eqeq",
+            [],
+            {"SEMGREP_SEND_METRICS": "auto"},
+            True,
+        ),
         ("rules/eqeq.yaml", [], {"SEMGREP_SEND_METRICS": "off"}, False),
-        ("r/a", [], {"SEMGREP_SEND_METRICS": "off"}, False),
+        (
+            "r/python.lang.correctness.useless-eqeq.useless-eqeq",
+            [],
+            {"SEMGREP_SEND_METRICS": "off"},
+            False,
+        ),
         ("rules/eqeq.yaml", [], {"SEMGREP_SEND_METRICS": "on"}, True),
-        ("r/a", [], {"SEMGREP_SEND_METRICS": "on"}, True),
+        (
+            "r/python.lang.correctness.useless-eqeq.useless-eqeq",
+            [],
+            {"SEMGREP_SEND_METRICS": "on"},
+            True,
+        ),
         (
             "rules/eqeq.yaml",
             ["--metrics", "auto"],
             {"SEMGREP_SEND_METRICS": "on"},
             False,
         ),
-        ("r/a", ["--metrics", "auto"], {"SEMGREP_SEND_METRICS": "off"}, True),
+        (
+            "r/python.lang.correctness.useless-eqeq.useless-eqeq",
+            ["--metrics", "auto"],
+            {"SEMGREP_SEND_METRICS": "off"},
+            True,
+        ),
     ],
 )
 def test_flags(
     run_semgrep_in_tmp, mock_config_request, config, options, env, should_send
 ):
+    """
+    Test that we try to send metrics when we should be
+    """
+    _, output = run_semgrep_in_tmp(
+        config,
+        options=[*options, "--debug"],
+        env={"SEMGREP_USER_AGENT_APPEND": "testing", **env},
+    )
+    print(output)
+    # Test that we try to send metrics. Even if it fails sending
+    assert (
+        "Sent pseudonymous metrics" in output
+        or "Failed to send pseudonymous metrics" in output
+        if should_send
+        else "Sent pseudonymous metrics" not in output
+    )
+
+
+@mark.parametrize(
+    "config,options,env,should_send",
+    [
+        ("rules/eqeq.yaml", [], {"SEMGREP_SEND_METRICS": "on"}, True),
+    ],
+)
+def test_flags_actual_send(
+    run_semgrep_in_tmp, mock_config_request, config, options, env, should_send
+):
+    """
+    Test that the server for metrics sends back success
```

Comment:
Just to confirm, this could still fail, but will likely be less flakey because the rulesets are shorter? From the PR, I got the idea that you were going to run only one of the options, but it looks like that didn't change. (Sorry if this comment is too soon I'm excited for this PR)

Reply:
Yep. Updated PR body